modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/websocket/WebSocketAPITopicsTestCase.java

```diff
@@ -0,0 +1,359 @@
+
+        URI endpointUri = new URI("ws://" + webSocketServerHost + ":" + webSocketServerPort);
+
+        apiRequest = new APIRequest(apiName, apiContext, endpointUri, endpointUri);
+        apiRequest.setVersion(apiVersion);
+        apiRequest.setTiersCollection(APIMIntegrationConstants.API_TIER.ASYNC_UNLIMITED);
+        apiRequest.setProvider(provider);
+        apiRequest.setType("WS");
+        HttpResponse addAPIResponse = restAPIPublisher.addAPI(apiRequest);
+        websocketAPIID = addAPIResponse.getData();
+        String asyncApiDefinition = "{" +
+                "   \"asyncapi\":\"2.0.0\"," +
+                "   \"info\":{" +
+                "      \"title\":\"" + apiRequest.getName() + "\"," +
+                "      \"version\":\"" + apiRequest.getVersion() +
+                "   \"}," +
+                "   \"servers\":{" +
+                "      \"production\":{" +
+                "          \"url\":\"" + endpointUri.toString() + "\"," +
+                "          \"protocol\":\"ws\"" +
+                "      }" +
+                "   }," +
+                "   \"channels\":{" +
+
+                // Topics with mappings
+                "      \"/test\":{" +
+                "          \"parameters\":{}," +
+                "          \"publish\":{" +
+                "              \"x-uri-mapping\":\"/test\"" +
+                "          }," +
+                "          \"subscribe\":{" +
+                "              \"x-uri-mapping\":\"/test\"" +
+                "          }" +
+                "      }," +
+                "      \"/test2/{prop}\":{" +
+                "          \"parameters\":{" +
+                "              \"prop\":{\"description\":\"\", \"schema\":{\"type\":\"string\"}}" +
+                "          }," +
+                "          \"publish\":{" +
+                "              \"x-uri-mapping\":\"/test2/{uri.var.prop}\"" +
+                "          }," +
+                "          \"subscribe\":{" +
+                "              \"x-uri-mapping\":\"/test2/{uri.var.prop}\"" +
+                "          }" +
+                "      }," +
+
+                // Topics without mappings
+                "      \"/test3\":{" +
+                "          \"parameters\":{}," +
+                "          \"publish\":{}," +
+                "          \"subscribe\":{}" +
+                "      }," +
+                "      \"/test3/{prop}\":{" +
+                "          \"parameters\":{}," +
+                "          \"publish\":{}," +
+                "          \"subscribe\":{}" +
+                "      }" +
+
+                "   }" +
+                "}";
+        restAPIPublisher.updateAsyncAPI(websocketAPIID, asyncApiDefinition);
+        restAPIPublisher.changeAPILifeCycleStatus(websocketAPIID, APILifeCycleAction.PUBLISH.getAction(), null);
+        waitForAPIDeploymentSync(user.getUserName(), apiName, apiVersion, APIMIntegrationConstants.IS_API_EXISTS);
```

Comment:
Let's add this line after 189

Reply:
Done in https://github.com/wso2/product-apim/pull/10992/commits/e2ce713e42731bc465c92045d3d537114895b337